SECEdgar/extractor/EDGARExtractor.py

```diff
@@ -0,0 +1,227 @@
+                                         + "." + str(sec_doc_num)
+                                         + "." + str(doc_num)
+                                         + "." + doc_filename_split[1])
+
+                # DEBUG ONLY: Write filedata to file
+                # filedata_filename = os.path.join(out_dir,
+                # in_filename + "." + str(doc_num) + ".data_" + filetype)
+                # with open(filedata_filename, "w", encoding="utf8") as outfh:
+                #    outfh.write(filedata)
+
+                # Is the file uuencoded?
+                is_uuencoded = filedata.find("begin 644 ") != -1
+
+                # File is uu-encoded
+                if is_uuencoded:
+                    logging.info("{} contains an uu-encoded file"
+                                 .format(in_file))
+                    encfn = outfn + ".uu"
+                    with open(encfn, "w", encoding="utf8") as encfh:
+                        encfh.write(filedata)
+                    uu.decode(encfn, outfn)
+                    os.remove(encfn)
+
+                # Plain file (no conversion needed)
+                else:
+                    logging.info("{} contains an non uu-encoded file"
+                                 .format(in_file))
+                    with open(outfn, "w", encoding="utf8") as outfh:
+                        outfh.write(filedata)
+            # doc-num loop end
+
+            # Save SEC-DOCUMENT metadata to file
+            with open(metadata_file, "w", encoding="utf8") as fileh:
+                fileh.write(self.jsonPretty(metadata))
+        # sec_doc_num loop end
+
+        if(rm_infile):
+            os.remove(in_file)
+
+    @staticmethod
+    def procKeyStr(s):
+        return s.replace(" ", "_")
+
+    @staticmethod
+    def jsonPretty(dict_data):
+        return json.dumps(dict_data, sort_keys=True,
+                          indent=4, separators=(',', ': '),
+                          ensure_ascii=False)
+
+    def procDocMetadata(self, doc):
+        metadata_doc = dict()
+
+        # Document type
+        type_m = re.search("<TYPE>(.*?)\n", doc)
+        if type_m:
+            metadata_doc["type"] = type_m.group(1)
+
+        # Document sequence
+        seq_m = re.search("<SEQUENCE>(.*?)\n", doc)
+        if seq_m:
+            metadata_doc["sequence"] = seq_m.group(1)
+
+        # Document filename
+        fn_m = re.search("<FILENAME>(.*?)\n", doc)
+        metadata_doc["filename"] = fn_m.group(1)
+
+        return metadata_doc
+
+    # Process the metadata of the focal document
+    def processMetadata(self, curr_doc):
+        out_dict = dict()
+        levels = [None, None]
+
+        for line in curr_doc.split("\n"):
+
+            if self.metadata_debug:
+                logging.debug("Line: '{}'".format(line))
+
+            if "<ACCEPTANCE-DATETIME>" in line:
+                out_dict["acceptance-datetime"] = \
+                    line[len("<ACCEPTANCE-DATETIME>"):]
```

Comment:
You have already put a lot of effort but I guess it needs a little bit of refinements and then we can add it to the project.